components/server/ee/src/gitlab/gitlab-app-support.ts

```diff
@@ -61,4 +61,18 @@ export class GitLabAppSupport {
         result.unshift(...ownersRepos);
         return result;
     }
+
+    protected async getAccountAvatarUrl(anyProject: any, providerHost: string): Promise<string> {
```

Comment:
Is it possible to do any better than an `any` type here?

Reply:
Good catch!

The original code uses `const anyProject = project as any`, so I've just used that in this method.

It looks like the types returned by the API are of type `ProjectSchema`, but also that type seems incomplete:

<img width="957" alt="Screenshot 2022-03-15 at 14 40 01" src="https://user-images.githubusercontent.com/599268/158390455-e603b461-95be-4f8f-a09a-5a1214e6c156.png">

Reply:
We could potentially reverse-engineer the type, e.g.:

```ts
type ProjectSchema = {
    owner?: {
        avatar_url: string;
    },
    namespace: {
        avatar_url: string;
    },
};
```

but I'm not sure if this is better than `any`. 🤔

Reply:
One option would be to get the data returned by the API and then use `typeof` to generate a type from it, if there is no 'official' specification available.